javers-core/src/test/groovy/org/javers/core/JaversDiffE2ETest.groovy

```diff
@@ -114,6 +114,35 @@ class JaversDiffE2ETest extends AbstractDiffTest {
         diff.changes.get(0).affectedGlobalId.value() == DummyCompositePoint.class.name+"/(1,2)"
     }
 
+    class DummyWithEntityId {
+        @Id EntityAsId entityAsId
+        int value
+    }
+
+    class EntityAsId {
+        @Id
+        int id
+        int value
+    }
+
+    def "should use nested Entity Id value as Id of parent Entity"(){
+        given:
+        def javers = javers().build()
```

Comment:
When you don't register EntityAsId as Entity, this test fails, because
EntityAsId is interred as Value. and when you don't touch DummyWithEntityId, EntityAsId would be inferred as Entity. This is not ok.

see how TypeFactory.inferIdPropertyTypeAsValue() works

Reply:
okay, I've fixed this issue in ` TypeFactory.inferIdPropertyTypeAsValue()`, also I've simplified the change in `localIdAsString()`

Reply:
I liked your changes. I didn't know about the core of Javers. I will study more about this. Thank you.